docs/deployment-patterns/deployment-as-code.md

```diff
@@ -0,0 +1,113 @@
+---
+title: Deployment as code
+description: You can define your entire Octopus configuration as code and track it all through source control. This page describes the different options available right now, and our plans to make this even better in the future.
+position: 4
+---
+
+You are probably reading this page because you want to manage your **deployment as code**. You want to define your deployment process, scripts, and variables in code, perhaps colocating this with the application source code in your git repository. This page describes the different options available in Octopus right now, and our plans to make this even better in the future.
+
+We recommend taking a two-phase approach when moving towards **deployment as code**:
+
+1. Start with [scripts as code](#scripts-as-code) - this method offers the best cost/reward ratio, is the simplest to implement and maintain over time.
+2. Move towards [project as code](#project-as-code) - this method is more difficult to set up, but comes with the benefit of having your deployment managed as code.
+
+## Scripts as code {#scripts-as-code}
+
+The simplest way to get started with **deployment as code** is to manage your custom deployment **scripts as code**. When you deploy your application, Octopus can execute a script contained inside a package. You can colocate your deployment scripts with your application source code, leveraging all the benefits of source control including change tracking and branching, then package it all up for Octopus.
+
+There is a downside to stopping here: your scripts are managed as code, but your deployment process and variables are still controlled by Octopus. Depending on your situation, this tradeoff might be quite acceptable.
+
+### Getting started
```

Comment:
While we're trying to reduce skimming people still will so having titles that state what's coming next will help catch their eye. Something like ### Move Your Scripts into Source Code

Reply:
Done